version: "3.9"
services:

  postgres:
    image: postgres:13.3
    environment:
      POSTGRES_DB: "test-db"
      POSTGRES_USER: "user"
      POSTGRES_PASSWORD: "pwd"
    volumes:
      - ./storage/migrations/init.sql:/docker-entrypoint-initdb.d/init.sql
      - pg-data:/var/lib/postgresql/data
    ports:
      - "5432:5432"
    networks:
      - assignment
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -d $${POSTGRES_DB} -U $${POSTGRES_USER}"]
      interval: 10s
      timeout: 3s
      retries: 3

  pgadmin:
    image: dpage/pgadmin4:7.2
    environment:
      PGADMIN_DEFAULT_EMAIL: "test@test.test"
      PGADMIN_DEFAULT_PASSWORD: "password"
      PGADMIN_CONFIG_SERVER_MODE: "False"
    volumes:
      - pgadmin-data:/var/lib/pgadmin
    ports:
      - "5050:80"
    networks:
      - assignment

volumes:
  pg-data:
  pgadmin-data:

networks:
  assignment:
    driver: bridge